### PR TITLE
Add a GPU section to the documentation

### PR DESCRIPTION
**Pull request by amontoison**
Code tested with P100 GPU.

### CODE REVIEW

File: docs/src/gpu.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+## GPU support
+
+All solvers in Krylov.jl can be used with `CuArrays` and allows computations with Nvidia GPU. Problems stored in CPU format (`Matrix` and `Vector`) must be converted to GPU format (`CuMatrix` and `CuVector`).
```
Comment (dpo) [inline docs/src/gpu.md:3]: allows -> allow

Comment (dpo) [reply docs/src/gpu.md:3]: ... must first be converted ...

File: docs/src/gpu.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+# GPU Arrays
+A_gpu = CuMatrix(A_cpu)
+b_gpu = CuVector(b_cpu)
+
+# Solve a square and dense system on GPU
+x, stats = dqgmres(A_gpu, b_gpu)
+```
+
+Sparse matrices also have a specific storage on GPU (`CuSparseMatrixCSC` or `CuSparseMatrixCSR`).
```
Comment (dpo) [inline docs/src/gpu.md:20]: remove "also" and finish sentence with ":"

**Comment by amontoison** [comment]
Thanks Dominique!

**Comment by amontoison** [comment]
Let's merge this too?

**Comment by dpo** [comment]
Let's do it.

**Pull request merged**